relay-general/src/store/normalize/spans.rs

```diff
@@ -44,14 +21,14 @@ fn merge_non_overlapping_intervals(intervals: &mut [TimeInterval]) -> Vec<TimeIn
 
     // The first interval is stored in `previous`, so make sure to skip it.
     for current in intervals.iter().skip(1) {
-        if current.end < previous.end {
+        if current.end_timestamp < previous.end_timestamp {
```

Comment:
The shorter names in this module were actually nice. If you like, feel free to change TimeWindowSpan's fields to `start` and `end`, up to you :)